io/pci/dlpar.py

```diff
@@ -0,0 +1,257 @@
+        output = self.listhwres(self.server, self.lpar_1, self.drc_index)
+        if self.drc_index not in output[0]:
+            self.log.debug(output)
+            self.fail("lshwres fails to list the drc after dlpar add")
+
+    def dlpar_move(self):
+        '''
+        dlpar move operation from lpar_1 to lpar2 and back from
+        lpar_2 to lpar_1
+        '''
+        if self.lpar_2 is None:
+            return
+
+        output = self.changehwres(self.server, 'm', self.lpar_id, self.lpar_2,
+                                  self.drc_index, 'move')
+
+        output = self.listhwres(self.server, self.lpar_1, self.drc_index)
+        if self.drc_index in output[0]:
+            self.log.debug(output)
+            self.fail("lshwres still lists the drc in lpar_1 after \
+                      dlpar move to lpar_2")
+
+        output = self.listhwres(self.server, self.lpar_2, self.drc_index)
+        if self.drc_index not in output[0]:
+            self.log.debug(output)
+            self.fail("lshwres fails to list the drc in lpar_2 after \
+                       dlpar move")
+
+        # dlpar move operation from lpar2 to lpar1
+        output = self.changehwres(self.server, 'm', self.lpar_id, self.lpar_1,
+                                  self.drc_index, 'move')
+
+        output = self.listhwres(self.server, self.lpar_1, self.drc_index)
+        if self.drc_index not in output[0]:
+            self.log.debug(output)
+            self.fail("lshwres fails to list the drc in lpar_1 after \
+                       dlpar move")
+
+        output = self.listhwres(self.server, self.lpar_2, self.drc_index)
+        if self.drc_index in output[0]:
+            self.log.debug(output)
+            self.fail("lshwres still lists the drc in lpar_2 after \
+                      dlpar move to lpar_1")
+
+    def listhwres(self, server, lpar, drc_index):
+        cmd = 'lshwres -r io -m %s \
+               --rsubtype slot --filter lpar_names= %s \
+               | grep -i %s' % (server, lpar, drc_index)
+        try:
+            cmd = self.run_command(cmd)
+        except CommandFailed as cf:
+            self.log.debug(str(cf))
+            self.fail("lshwres operation failed ")
+        return cmd
+
+    def changehwres(self, server, operation, lpar_id, lpar, drc_index, msg):
+        if operation == 'm':
+            cmd = 'chhwres -r io --rsubtype slot -m %s \
+               -o %s --id %s -t %s -l %s ' % (server, operation, lpar_id,
+                                              lpar, drc_index)
+        else:
+            cmd = 'chhwres -r io --rsubtype slot -m %s \
+                   -o %s --id %s -l %s ' % (server, operation, lpar_id,
+                                            drc_index)
+        try:
+            cmd = self.run_command(cmd)
+        except CommandFailed as cf:
+            self.log.debug(str(cf))
+            self.fail("dlpar %s operation failed" % msg)
+
+        return cmd
```

Comment:
Sorry for the late comment, but this function does not need to return anything. The caller does not do anything with the returned output.
```
try:
    self.run_command(cmd)
except ...:
    ...
    ..
```